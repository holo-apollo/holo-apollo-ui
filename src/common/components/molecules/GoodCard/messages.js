import { defineMessages } from 'react-intl';

export default defineMessages({
  by: {
    id: 'GoodCard.by',
    defaultMessage: 'by',
  },
  purchase: {
    id: 'GoodCard.purchase',
    defaultMessage: 'Purchase',
  },
});
