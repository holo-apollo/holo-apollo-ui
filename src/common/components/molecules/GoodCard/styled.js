// @flow
import styled, { css } from 'styled-components';

import palette from 'common/palette';

const getContStyles = ({ hovered }: { hovered: boolean }) => css`
  border: solid 1px ${hovered ? palette.darkGrey : palette.lightGrey};
`;

export const Cont = styled.div`
  ${getContStyles};
`;

const getImgContStyles = ({
  imgUrl,
  hovered,
}: {
  imgUrl: string,
  hovered: boolean,
}) => css`
  background-image: url(${imgUrl});
  border-bottom: solid 1px ${hovered ? palette.darkGrey : palette.lightGrey};
`;

export const ImgCont = styled.div`
  height: 261px;
  background-repeat: no-repeat;
  background-position: center;
  background-size: cover;
  ${getImgContStyles};
`;

export const ContentCont = styled.div`
  height: 86px;
  padding: 12px;
  position: relative;
`;

export const NameCont = styled.div`
  font-size: 12px;
  font-weight: 500;
  color: ${palette.grey};
  margin-bottom: 10px;
`;

export const StoreNameCont = styled.div`
  font-size: 8px;
  font-weight: 300;
  color: ${palette.grey};
  margin-bottom: 10px;
`;

const getActionsContStyles = ({ visible }: { visible: boolean }) => css`
  display: ${visible ? 'initial' : 'none'};
`;

export const ActionsCont = styled.div`
  position: absolute;
  bottom: 12px;
  right: 12px;
  ${getActionsContStyles};
`;
