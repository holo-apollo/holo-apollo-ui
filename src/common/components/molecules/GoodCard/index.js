// @flow
import React from 'react';
import { FormattedMessage } from 'react-intl';
import { withState } from 'recompose';
import Link from 'next/link';

import Button from 'common/components/buttons/Button';
import PriceLabel from 'common/components/molecules/PriceLabel';
import { getGoodPageLink } from 'helpers/urls';
import {
  Cont,
  ImgCont,
  ContentCont,
  NameCont,
  StoreNameCont,
  ActionsCont,
} from './styled';
import messages from './messages';

type Props = {
  id: number,
  name: string,
  mainImageUrl: string,
  price: number,
  priceCurrency: string,
  discount: number,
  seller: {
    storeName: string,
  },
  onPurchase: () => void,
  hovered: boolean,
  setHovered: boolean => void,
};

const PureGoodCard = ({
  id,
  name,
  mainImageUrl,
  price,
  priceCurrency,
  discount,
  seller,
  onPurchase,
  hovered,
  setHovered,
}: Props) => (
  <Cont
    onMouseOver={() => setHovered(true)}
    onMouseOut={() => setHovered(false)}
    hovered={hovered}
  >
    <Link href={getGoodPageLink(id)}>
      <a>
        <ImgCont imgUrl={mainImageUrl} hovered={hovered} />
      </a>
    </Link>
    <ContentCont>
      <Link href={getGoodPageLink(id)}>
        <a>
          <NameCont>{name}</NameCont>
        </a>
      </Link>
      <StoreNameCont>
        <FormattedMessage {...messages.by} />: {seller.storeName}
      </StoreNameCont>
      <PriceLabel
        price={price}
        priceCurrency={priceCurrency}
        discount={discount}
      />
      <ActionsCont visible={hovered}>
        <Button width={110} fontWeight="normal" onClick={onPurchase}>
          <FormattedMessage {...messages.purchase} />
        </Button>
      </ActionsCont>
    </ContentCont>
  </Cont>
);

// $FlowFixMe
const GoodCard = withState('hovered', 'setHovered', false)(PureGoodCard);

export default GoodCard;
