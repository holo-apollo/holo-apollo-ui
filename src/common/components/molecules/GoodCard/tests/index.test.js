import React from 'react';
import { mount } from 'enzyme';
import renderer from 'react-test-renderer';

import GoodCard from '../index';

describe('GoodCard', function() {
  it('should render normally', function() {
    const wrapper = mount(
      <GoodCard
        name="Flower in a bowl"
        mainImageUrl="http://flowersandmorebyerin.com/wp-content/uploads/2015/02/pFlower3.png"
        price={95}
        priceCurrency="UAH"
        sellerInfo={{ storeName: 'Gold stuff shopping' }}
        onPurchase={jest.fn()}
      />
    );
    expect(wrapper.exists()).toBe(true);
  });

  it('should match snapshot', function() {
    const tree = renderer
      .create(
        <GoodCard
          name="Flower in a bowl"
          mainImageUrl="http://flowersandmorebyerin.com/wp-content/uploads/2015/02/pFlower3.png"
          price={95}
          priceCurrency="UAH"
          sellerInfo={{ storeName: 'Gold stuff shopping' }}
          onPurchase={jest.fn()}
        />
      )
      .toJSON();
    expect(tree).toMatchSnapshot();
  });
});
