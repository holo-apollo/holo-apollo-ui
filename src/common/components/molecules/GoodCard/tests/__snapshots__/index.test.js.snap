// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`GoodCard should match snapshot 1`] = `
<div
  className="styled__Cont-sc-1jbl3we-0 hLtCzR"
  onMouseOut={[Function]}
  onMouseOver={[Function]}
>
  <a
    href="/goods/1"
    onClick={[Function]}
  >
    <div
      className="styled__ImgCont-sc-1jbl3we-1 fPukLE"
    />
  </a>
  <div
    className="styled__ContentCont-sc-1jbl3we-2 egjtzB"
  >
    <a
      href="/goods/1"
      onClick={[Function]}
    >
      <div
        className="styled__NameCont-sc-1jbl3we-3 iccvMk"
      >
        Flower in a bowl
      </div>
    </a>
    <div
      className="styled__StoreNameCont-sc-1jbl3we-4 jrIYBS"
    >
      <span>
        by
      </span>
      : 
      Gold stuff shopping
    </div>
    <div
      className="styled__Cont-sc-1rmxlbk-0 kopYzg"
    >
      <div
        className="styled__PriceCont-sc-1rmxlbk-1 jaXxZL"
        size="s"
      >
        NaN
         
        UAH
      </div>
    </div>
    <div
      className="styled__ActionsCont-sc-1jbl3we-5 crzbcu"
    >
      <button
        className="Button__StyledButton-kpekc2-0 fnePeO"
        color="#424242"
        fontSize={14}
        fontWeight="normal"
        height={35}
        onClick={[Function]}
        width={110}
      >
        <span>
          Purchase
        </span>
      </button>
    </div>
  </div>
</div>
`;
